Jenkinsfile

```diff
@@ -125,6 +123,29 @@ pipeline {
                     }
                     sh "env; echo Building BUILD_IMAGE=${env.BUILD_IMAGE} for BUILD_TARGET=${env.BUILD_TARGET}, CHANGE_ID=${env.CHANGE_ID}, CACHEBUST=${env.CACHEBUST}."
                     sh "docker build --cache-from ${env.BUILD_IMAGE} --label cachebust=${env.CACHEBUST} --target build --build-arg CACHEBUST=${env.CACHEBUST} -t ${env.BUILD_IMAGE} ."
+                    // Create a running container (https://stackoverflow.com/a/38308399).
+                    sh "docker create --name=builder -it ${env.BUILD_IMAGE}"
+                    sh "docker start builder"
+
+                    // Allow non-root users to become root via sudo when they are part of the root group.
+                    // ${DockerBuildArgs()} includes the necessary parameters for that.
+                    sh "docker exec builder mkdir /etc/sudoers.d"
+                    sh "echo '%root ALL=(ALL) NOPASSWD: ALL' | docker exec -i builder tee /etc/sudoers.d/nopasswd >/dev/null"
+
+                    // sudo needs a user entry for the jenkins user.
+                    sh "echo jenkins:x:`id -u`:0:Jenkins:`pwd`/..:/bin/bash | docker exec -i builder tee --append /etc/passwd >/dev/null"
+                    sh "echo 'jenkins:*:0:0:99999:0:::' | docker exec -i builder tee --append /etc/shadow >/dev/null"
+
+                    // Now commit those changes for use below and clean up.
+                    sh "docker commit builder ${env.BUILD_IMAGE}"
+                    sh "docker stop builder"
+                    sh "docker rm builder"
+
+                    // Verify that docker and mount work under sudo in the updated image (needed for etcd tmpfs, see start-kubernetes.sh).
+                    // sudo depends on --privileged.
+                    sh "docker run --rm ${DockerBuildArgs()} ${env.BUILD_IMAGE} docker ps"
```

Comment:
I guess these below steps are added as part of debugging, I think we can remove these checks from the final version?

Reply:
I kept them intentionally because these are fast, local operations which ensure that a CI job gets aborted early if (for whatever reason) these crucial aspects of the build container no longer work.